include/aws/s3/s3_endpoint_resolver/s3_endpoint_resolver.h

```diff
@@ -0,0 +1,49 @@
+#ifndef AWS_S3_ENDPOINT_RESOLVER_H
+#define AWS_S3_ENDPOINT_RESOLVER_H
+
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/s3/s3.h>
+#include <aws/sdkutils/endpoints_rule_engine.h>
```

Comment:
```suggestion
```

in public headers,  avoid `#include`ing other headers, prefer to forward-declare anywhere you can get away with it

it keeps build times down for us and our users now and forever

Reply:
Thanks, fixed.